internal/controller/frrconfiguration_controller.go

```diff
@@ -80,7 +84,17 @@ func (r *FRRConfigurationReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, err
 	}
 
-	config, err := apiToFRR(cfgs)
+	secrets, err := r.getSecrets(ctx)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	config, err := apiToFRR(cfgs, secrets)
+	secretNotFound := SecretNotFoundError{}
+	if errors.As(err, &secretNotFound) { // This might be a receiving order issue so we need to return the error and retry
```

Comment:
is this necessary now that we listen for secret events?

Reply:
I'd rather be on the paranoid side. It should converge anyway